### PR TITLE
#739: removed "not inside an IDE installation message"

### PR DESCRIPTION
**Pull request by leonrohne27**
### Implements:

- removed the message "you are not inside an IDE installation"

### CODE REVIEW

**Review by jan-vcapgemini — APPROVED** [review]
Thanks, LGTM. Ready for review.

File: cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java]: This is incorrect IMHO.
If I call a commandlet that requires IDE_HOME the user needs to still get this error message to understand that his command failed.
```
$ ide install mvn
You are not inside an IDE installation: D:\projects

Error: IDEasy failed with exit code 2
```
This is still correct. However, this should be changed
```
$ ide
You are not inside an IDE installation: D:\projects
```

File: cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java:824]: You should not remove this.
It will cause strange errors otherwise and therefore this handling must stay.

Comment (leonrohne27) [reply cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java]: Thanks, I readded it

File: cli/src/main/java/com/devonfw/tools/ide/commandlet/EnvironmentCommandlet.java
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/commandlet/EnvironmentCommandlet.java:57]: IDE_ROOT will always be defined after proper setup of IDEasy.
It is IDE_HOME that may not be defined and that is why you changed `isIdeHomeRequired()` above (there is also isIdeRootRequired()).
```suggestion
    if (context.getIdeHome() == null) {
```

File: cli/src/main/java/com/devonfw/tools/ide/cli/NullIdeRootException.java
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/cli/NullIdeRootException.java:6]: If you really want to express what you stated in the JavaDoc, it would be better to rename it to `IdeRootIsNullException` so one can better understand and guess what all this is about.
However, an exception that we use to abort without any message should not be named after the condition it was caused by but rather than the problem type it represents and potential handling it may imply.
Therefore I would suggest renaming to `CliCancelException` or `CliExitException`.
Also it would be good if you define a custom exit code like already done in `CliAbortException`.

Comment (leonrohne27) [reply cli/src/main/java/com/devonfw/tools/ide/cli/NullIdeRootException.java]: done

**Review by hohwille — APPROVED** [review]
@leonrohne27 thanks for the rework. Nice job 👍 
Ready for merge.

**Comment by hohwille** [comment]
CHANGELOG entry is still missing. Could you quickly add that?
Otherwise I will merge in ~1h and add it after the merge...

**Pull request merged**
